### PR TITLE
binarycaching.md: Move Configuration section to the top. Add table of…

### PR DESCRIPTION
**Pull request by autoantwort**
… contents. Make clear that file location must be absolute. Add that read is the default mode.

**Describe the pull request**

- What does your PR fix?  
  The most important part should be at the top, add that the file location must be absolute and that the "read" mode is the default one.

### CODE REVIEW

File: docs/users/binarycaching.md [hunk 1]
```diff
@@ -146,38 +188,7 @@ Commas (`,`) are valid as part of a object prefix in GCS, just remember to escap
...
-| `nuget,<uri>[,<rw>]`        | Adds a NuGet-based source; equivalent to the `-Source` parameter of the NuGet CLI
-| `nugetconfig,<path>[,<rw>]` | Adds a NuGet-config-file-based source; equivalent to the `-Config` parameter of the NuGet CLI. This config should specify `defaultPushSource` for uploads.
-| `x-azblob,<baseuri>,<sas>[,<rw>]`    | **Experimental: will change or be removed without warning**<br> Adds an Azure Blob Storage source. Uses Shared Access Signature validation. URL should include the container path.
-| `interactive`               | Enables interactive credential management for NuGet (for debugging; requires `--debug` on the command line)
-
-The `<rw>` optional parameter for certain sources controls whether they will be consulted for
-downloading binaries (`read`), whether on-demand builds will be uploaded to that remote (`write`), or both (`readwrite`).
-
-### Nuget Provider Configuration
+## Nuget Provider Configuration
```
Comment (ras0219) [inline docs/users/binarycaching.md:191]:
```suggestion
## NuGet Provider Configuration
```

File: docs/users/binarycaching.md [hunk 2]
```diff
@@ -6,7 +6,49 @@ Binary caching is vcpkg's method for reusing package builds between projects and
...
+In-tool help is available via `vcpkg help binarycaching`.
+
+Table of Contents
+ - [Configuration](#configuration)
+ - [CI Examples](#ci-examples)
+   - [GitHub Packages](#github-packages)
+   - [Azure DevOps Artifacts](#azure-devops-artifacts)
+   - [Azure Blob Storage](#azure-blob-storage-experimental)
+   - [Google Cloud Storage](#google-cloud-storage-experimental)
+ - [Nuget Provider Configuration](#nuget-provider-configuration)
```
Comment (ras0219) [inline docs/users/binarycaching.md:18]:
```suggestion
 - [NuGet Provider Configuration](#nuget-provider-configuration)
```

File: docs/users/binarycaching.md [hunk 3]
```diff
@@ -6,7 +6,49 @@ Binary caching is vcpkg's method for reusing package builds between projects and
...
+| `clear`                     | Removes all previous sources (including the default)
+| `default[,<rw>]`            | Adds the default file-based location
+| `files,<absolute path>[,<rw>]`       | Adds a custom file-based location
+| `nuget,<uri>[,<rw>]`        | Adds a NuGet-based source; equivalent to the `-Source` parameter of the NuGet CLI
+| `nugetconfig,<path>[,<rw>]` | Adds a NuGet-config-file-based source; equivalent to the `-Config` parameter of the NuGet CLI. This config should specify `defaultPushSource` for uploads.
+| `x-azblob,<baseuri>,<sas>[,<rw>]`    | **Experimental: will change or be removed without warning**<br> Adds an Azure Blob Storage source. Uses Shared Access Signature validation. URL should include the container path.
+| `interactive`               | Enables interactive credential management for NuGet (for debugging; requires `--debug` on the command line)
+
+The `<rw>` optional parameter for certain sources controls whether they will be consulted for
+downloading binaries (`read`)(default), whether on-demand builds will be uploaded to that remote (`write`), or both (`readwrite`).
```
Comment (ras0219) [inline docs/users/binarycaching.md:51]:
```suggestion
downloading binaries (`read`)(default), whether on-demand builds will be uploaded to that remote (`write`), or both (`readwrite`).

Additional configuration details for NuGet-based providers can be found below in [NuGet Provider Configuration](#nuget-provider-configuration).
```

File: docs/users/binarycaching.md [hunk 4]
```diff
@@ -146,38 +188,7 @@ Commas (`,`) are valid as part of a object prefix in GCS, just remember to escap
...
-| `x-azblob,<baseuri>,<sas>[,<rw>]`    | **Experimental: will change or be removed without warning**<br> Adds an Azure Blob Storage source. Uses Shared Access Signature validation. URL should include the container path.
-| `interactive`               | Enables interactive credential management for NuGet (for debugging; requires `--debug` on the command line)
-
-The `<rw>` optional parameter for certain sources controls whether they will be consulted for
-downloading binaries (`read`), whether on-demand builds will be uploaded to that remote (`write`), or both (`readwrite`).
-
-### Nuget Provider Configuration
+## Nuget Provider Configuration
 
 #### Credentials
```
Comment (ras0219) [inline docs/users/binarycaching.md:193]: The headers in this section need to be adjusted for the changes in `Nuget Provider Configuration`
```suggestion
### Credentials
```

**Review by ras0219 — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks!

**Pull request merged**